// react-native-material-datetime-picker/lib/module/datepicker.android.js
import { NativeModules } from 'react-native';
import { LINKING_ERROR } from './utils/constants';
const RNMaterialDatePicker = NativeModules.RNMaterialTimePicker ? NativeModules.RNMaterialDatePicker : new Proxy({}, {
  get() {
    throw new Error(LINKING_ERROR);
  }

});
export default class DatePicker {
  /**
   * Shows the Android Material Design time picker dialog.
   */
  static async show(options) {
    var _options$minimumDate, _options$maximumDate, _options$startDate, _options$endDate;

    return RNMaterialDatePicker.show({ ...options,
      value: options.value.getTime(),
      minimumDate: (_options$minimumDate = options.minimumDate) === null || _options$minimumDate === void 0 ? void 0 : _options$minimumDate.getTime(),
      maximumDate: (_options$maximumDate = options.maximumDate) === null || _options$maximumDate === void 0 ? void 0 : _options$maximumDate.getTime(),
      startDate: (_options$startDate = options.startDate) === null || _options$startDate === void 0 ? void 0 : _options$startDate.getTime(),
      endDate: (_options$endDate = options.endDate) === null || _options$endDate === void 0 ? void 0 : _options$endDate.getTime()
    });
  }

  static async dismiss() {
    return RNMaterialDatePicker.dismiss();
  }

}
