// react-native-material-datetime-picker/lib/module/types.js
export let AndroidPickerMode;

(function (AndroidPickerMode) {
  AndroidPickerMode["DATE"] = "date";
  AndroidPickerMode["TIME"] = "time";
})(AndroidPickerMode || (AndroidPickerMode = {}));

export let AndroidTimeInputMode;

(function (AndroidTimeInputMode) {
  AndroidTimeInputMode["DEFAULT"] = "default";
  AndroidTimeInputMode["KEYBOARD"] = "keyboard";
  AndroidTimeInputMode["CLOCK"] = "clock";
})(AndroidTimeInputMode || (AndroidTimeInputMode = {}));

export let AndroidDateInputMode;

(function (AndroidDateInputMode) {
  AndroidDateInputMode["DEFAULT"] = "default";
  AndroidDateInputMode["TEXT"] = "text";
  AndroidDateInputMode["CALENDAR"] = "calendar";
})(AndroidDateInputMode || (AndroidDateInputMode = {}));

export let AndroidDatePickerType;

(function (AndroidDatePickerType) {
  AndroidDatePickerType["DEFAULT"] = "default";
  AndroidDatePickerType["RANGE"] = "range";
})(AndroidDatePickerType || (AndroidDatePickerType = {}));

export let ActionType;

(function (ActionType) {
  ActionType[ActionType["SET_DATE"] = 0] = "SET_DATE";
  ActionType[ActionType["SET_DATE_RANGE"] = 1] = "SET_DATE_RANGE";
  ActionType[ActionType["SET_TIME"] = 2] = "SET_TIME";
  ActionType[ActionType["DISMISSED"] = 3] = "DISMISSED";
})(ActionType || (ActionType = {}));
