// @tighten/react-native-time-input/lib/module/components/TimeTextField.js
import React, { useEffect, useState } from 'react';
import { TextInput } from 'react-native';
import * as TimeInputHelper from '../helpers/timeInput';
import useDebounce from '../hooks/useDebounce';
export default function TimeTextField({
  givenTime,
  onTimeValueReady,
  style
}) {
  const [time, setTime] = useState('');
  const {
    state: debouncedTime,
    setDebouncedState: setDebouncedTime,
    debounce
  } = useDebounce(time, 250);
  useEffect(() => {
    if (!givenTime) return;
    setTime(givenTime.time);
  }, [givenTime, setTime]);
  useEffect(() => {
    onTimeValueReady(TimeInputHelper.validate(debouncedTime), debouncedTime);
  }, [debouncedTime, onTimeValueReady]);
  useEffect(() => {
    setDebouncedTime(time);
    return () => {
      debounce.cancel();
    };
  }, [debounce, time, setDebouncedTime]);
  return /*#__PURE__*/React.createElement(TextInput, {
    keyboardType: "number-pad",
    maxLength: 5,
    onChangeText: text => setTime(TimeInputHelper.mask(text)),
    placeholder: "08:00",
    value: time,
    style: style
  });
}
